pkg/api/dataproxy.go

```diff
@@ -107,6 +108,17 @@ func ProxyDataSourceRequest(c *middleware.Context) {
 
 	proxyPath := c.Params("*")
 
+	if ds.Type == m.DS_PROMETHEUS {
+		if c.Req.Request.Method != "GET" {
+			c.JsonApiErr(403, "GET is only allowed on proxied Prometheus datasource", nil)
+			return
+		}
+		if c.Req.Request.Method == "GET" && strings.Index(proxyPath, "/api/") == 0 {
```

Comment:
ok, but why then check method GET and if proxyPath starts with /api/   ? 

The method is already checked in the previous if statement so it seems redundant. Maybe combine to one if statement? 

if c.Req.Request.Method != "GET" || strings.Index(proxyPath, "/api/") == 0